types/node/ts3.7/assert.d.ts

```diff
@@ -0,0 +1,53 @@
+declare module "assert" {
+    function internal(value: any, message?: string | Error): asserts value;
+    namespace internal {
+        class AssertionError implements Error {
+            name: string;
+            message: string;
+            actual: any;
+            expected: any;
+            operator: string;
+            generatedMessage: boolean;
+            code: 'ERR_ASSERTION';
+
+            constructor(options?: {
+                message?: string; actual?: any; expected?: any;
+                operator?: string; stackStartFn?: Function
+            });
+        }
+
+        type AssertPredicate = RegExp | (new() => object) | ((thrown: any) => boolean) | object | Error;
+
+        function fail(message?: string | Error): never;
```

Comment:
Can we omit all the functions that do not use the `assert` functionality, TS should fall back iirc.

Reply:
It'd be great if it does, but sadly I can't get them working at all right now :(

Judging by the types in `ts3.5`, I'd say TS does fallback.

Reply:
That's odd, partial augmentation usually works, see `fs.d.ts` etc. in the ts3.5 folder 👀

Reply:
It's not the partial - it's just the whole thing entire.

It's actually weird, as TS seems to completely ignore `assert.d.ts` unless I move it into `ts3.5`, at which point it starts complaining about `export = internal` being a duplicate.

It's like it's actually completely ignoring the `ts3.7` in favor of `ts3.5` - I wonder if it's FIFO?

Reply:
You need to put `ts3.7` before `ts3.5` in the `typesVersions` field.